core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -181,6 +185,34 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
       else consumer.pause(java.util.Collections.singleton(tp))
     }
 
+    def tryPoll(timeout: Long, onSuccess: ConsumerRecords[K, V] => Unit): Unit = {
+      Try(consumer.poll(timeout))
+        .transform(
+          rawResult => {
+            wakeupTask.cancel()
+            wakeups = 0
+            Success(onSuccess(rawResult))
+          },
+          f => {
+            wakeupTask.cancel()
+            Failure(f)
+          }
+        )
+        .recover {
+          case w: WakeupException =>
+            wakeups = wakeups + 1
+            if (wakeups == MaxWakeups) {
+              log.error("WakeupException limit exceeded, stopping")
+              context.stop(self)
+            }
+            else
+              log.warning("Consumer interrupted with WakeupException after timeout")
```

Comment:
include the `w.getMessage`, perhaps also mention the name of the timeout config and its current value